pkg/domain/infra/tunnel/images.go

```diff
@@ -288,3 +305,64 @@ func (ir *ImageEngine) Shutdown(_ context.Context) {
 func (ir *ImageEngine) Sign(ctx context.Context, names []string, options entities.SignOptions) (*entities.SignReport, error) {
 	return nil, errors.New("not implemented yet")
 }
+
+// Sourced from openshift image builder
+
+// TransformFileFunc is given a chance to transform an arbitrary input file.
+type TransformFileFunc func(h *tar.Header, r io.Reader) (data []byte, update bool, skip bool, err error)
+
+// filterArchive transforms the provided input archive to a new archive,
+// giving the fn a chance to transform arbitrary files.
+func filterArchive(r io.Reader, w io.Writer, fn TransformFileFunc) error {
+	tr := tar.NewReader(r)
+	tw := tar.NewWriter(w)
+
+	for {
+		h, err := tr.Next()
+		if err == io.EOF {
+			return tw.Close()
+		}
+		if err != nil {
+			return err
+		}
+
+		var body io.Reader = tr
+		name := h.Name
```

Comment:
From archive/tar documentation:

> For forward compatibility, users that retrieve a Header from Reader.Next, mutate it in some ways, and then pass it back to Writer.WriteHeader should do so by creating a new Header and copying the fields that they are interested in preserving.

Reply:
im seriously hesitant to fiddle with with this as it was taken directly from imagebuilders code, which is presumably tried and true?